examples/abitest/module/rust/src/lib.rs

```diff
@@ -479,35 +503,44 @@ impl FrontendNode {
     }
 
     fn test_channel_wait_raw(&self) -> std::io::Result<()> {
-        let (out_channel, in_channel) = oak::channel_create().unwrap();
-
-        // Write a message to the channel so wait operations don't block.
-        let data = vec![0x01, 0x02, 0x03];
-        expect_eq!(OakStatus::OK, oak::channel_write(out_channel, &data, &[]));
+        let mut out_channel = 0u64;
+        let mut in_channel = 0u64;
 
         unsafe {
+            oak::wasm::channel_create(&mut out_channel as *mut u64, &mut in_channel as *mut u64)
+                as i32;
+
+            // Write a message to the channel so wait operations don't block.
+            let data = vec![0x01, 0x02, 0x03];
+            expect_eq!(
+                OakStatus::OK.value() as u32,
+                oak::wasm::channel_write(
+                    out_channel,
+                    data.as_ptr(),
+                    data.len(),
+                    &[] as *const u8,
+                    0,
+                )
+            );
+
             expect_eq!(
                 OakStatus::ERR_INVALID_ARGS.value() as u32,
                 oak::wasm::wait_on_channels(invalid_raw_offset() as *mut u8, 1)
             );
         }
 
         unsafe {
-            // Wait on [write handle, ready read handle, empty read handle].
-            const COUNT: usize = 3;
+            // Wait on [write handle, ready read handle].
+            const COUNT: usize = 2;
             let mut space = Vec::with_capacity(COUNT * oak::wasm::SPACE_BYTES_PER_HANDLE);
             space
-                .write_u64::<byteorder::LittleEndian>(out_channel.handle)
+                .write_u64::<byteorder::LittleEndian>(out_channel)
                 .unwrap();
             space.push(0x00);
             space
-                .write_u64::<byteorder::LittleEndian>(in_channel.handle)
+                .write_u64::<byteorder::LittleEndian>(in_channel)
                 .unwrap();
             space.push(0x00);
-            space
-                .write_u64::<byteorder::LittleEndian>(self.backend_in[0].handle)
-                .unwrap();
-            space.push(0x99); // deliberate nonsense status value
```

Comment:
Why is this part of the test case dropped?

Reply:
`backend_in` contains `oak::Handle`s which now cannot be serialised, `sdk/rust/oak` doesn't export a method for getting their u64 value. I think this was original intended as a erroneous value that should fail? I could:

- replace it with a different erroneous value
- use `oak_runtime::Handle`s for backend_in

But it seems that making `oak::Handle` opaque and non-serialisable has removed the need for the current definition (it makes it impossible!).

Reply:
(The other handles in `write_u64` calls above have been changed to the `oak_runtime::Handle` which is still transparent)

Reply:
IIRC the test case was intended to get a variety of different channel read status values back, so if there were another channel available that would return `NOT_READY` then I think the same ground would be covered.